File: server/Simulation/Components/ChildOffsetComponent.cs
using System.Numerics;
using server.ECS;

namespace server.Simulation.Components;

[Component]
public struct ChildOffsetComponent(NTT EntityId, int parentId, Vector2 offset, float rotationOffset)
{
        public readonly NTT EntityId = EntityId;
        public readonly int ParentId = parentId;
        public readonly Vector2 Offset = offset;
        internal readonly float Rotation = rotationOffset;
}

File: pixiejsClient/src/app/ecs/components/ChildOffsetComponent.ts
import { Component } from "../core/Component";
import { Vector2 } from "../core/types";

export class ChildOffsetComponent extends Component {
  public offset: Vector2 = { x: 0, y: 0 };
  public rotation = 0;

  constructor(entityId: string, offset: Vector2, rotation: number) {
    super(entityId);
    this.offset = offset;
    this.rotation = rotation;
  }
}
